pkgs/development/libraries/physics/geant4/datasets.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv, fetchurl, writeTextFile }:
+
+let
+  mkDataset = { name, version, sha256, envvar}:
+    stdenv.mkDerivation {
+      inherit name version;
+
+      src = fetchurl {
+        url = "https://geant4-data.web.cern.ch/geant4-data/datasets/${name}.${version}.tar.gz";
+        inherit sha256;
+      };
+
+      preferLocalBuild = true;
+
+      installPhase = ''
+        mkdir -p $out/data
+        cp -R ./* $out/data
+      '';
+
+      inherit envvar;
+      setupHook = writeTextFile {
```

Comment:
I have my doubts about this. I see that ```writeTextFile``` is not used for static files in ```pkgs/```, I wonder if it would be better to stick with regular file here as well.

Reply:
I figured since the setup hooks were to small they could just be inlined. Is `writeTextFile` meant to be used in another way?

Reply:
I would not be surprised if there is some nasty corner case with it. I suppose I know one relatively harmless side-effect: if I'm not mistaken, the ```writeTextFile``` is a derivation and will "build" a new file for each architecture.

Reply:
I kind of agree with you, but that discussion goes beyond the scope of this PR.